src/System.Net.WebProxy/src/System/Net/WebProxy.cs

```diff
@@ -0,0 +1,238 @@
+                return true;
+            }
+
+            // If it matches the primary domain, it's local.  (Whether or not the hostname matches.)
+            string local = "." + IPGlobalProperties.GetIPGlobalProperties().DomainName;
+            return
+                local.Length == (hostString.Length - dot) &&
+                string.Compare(local, 0, hostString, dot, local.Length, StringComparison.OrdinalIgnoreCase) == 0;
+        }
+
+        private static bool IsAddressLocal(IPAddress ipAddress)
+        {
+            // TODO: The .NET Framework caches this and then uses network change
+            // notifications to track whether the set should be recomputed.
+            IPAddress[] localAddresses = Dns.GetHostEntryAsync(Dns.GetHostName()).GetAwaiter().GetResult().AddressList; // TODO: Use synchronous GetHostEntry when available
+
+            for (int i = 0; i < localAddresses.Length; i++)
+            {
+                if (ipAddress.Equals(localAddresses[i])) // TODO: was compare scope ID == false
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        public bool IsBypassed(Uri host)
+        {
+            if (host == null)
+            {
+                throw new ArgumentNullException(nameof(host));
+            }
+
+            return
+                Address == null ||
+                host.IsLoopback ||
+                (BypassProxyOnLocal && IsLocal(host)) ||
+                IsMatchInBypassList(host);
+        }
+
+        protected WebProxy(SerializationInfo serializationInfo, StreamingContext streamingContext)
+        {
+            BypassProxyOnLocal = serializationInfo.GetBoolean("_BypassOnLocal");
+            Address = (Uri)serializationInfo.GetValue("_ProxyAddress", typeof(Uri));
+            _bypassList = (ArrayList)serializationInfo.GetValue("_BypassList", typeof(ArrayList));
+            UseDefaultCredentials = serializationInfo.GetBoolean("_UseDefaultCredentials");
+        }
+
+        void ISerializable.GetObjectData(SerializationInfo serializationInfo, StreamingContext streamingContext)
+        {
+            GetObjectData(serializationInfo, streamingContext);
+        }
+
+        protected virtual void GetObjectData(SerializationInfo serializationInfo, StreamingContext streamingContext)
+        {
+            serializationInfo.AddValue("_BypassOnLocal", BypassProxyOnLocal);
+            serializationInfo.AddValue("_ProxyAddress", Address);
+            serializationInfo.AddValue("_BypassList", _bypassList);
+            serializationInfo.AddValue("_UseDefaultCredentials", UseDefaultCredentials);
+        }
+
+        [Obsolete("This method has been deprecated. Please use the proxy selected for you by default. http://go.microsoft.com/fwlink/?linkid=14202")]
+        public static WebProxy GetDefaultProxy()
+        {
+            // The .NET Framework here fetches returns a proxy that fetches IE settings and
```

Comment:
grammar:  "The .NET Framework here fetches returns" -> "The .NET Framework here returns"